src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -282,5 +283,73 @@ private static IEnumerable<JsonConverter> DefaultSimpleConverters
                 yield return new UriConverter();
             }
         }
+
+        // Get converter from the cached key converters or add a new Enum converter.
+        internal JsonConverter GetOrAddKeyConverter(Type keyType, Type dictionaryType)
+        {
+            if (s_keyConverters.TryGetValue(keyType, out JsonConverter? converter))
+            {
+                return converter;
+            }
+            // short-term solution, instead of using a factory pattern like JsonStringEnumConverter,
+            // we just create the enum converter here add we add it to s_keyConverters.
+            else if (keyType.IsEnum)
+            {
+                converter = CreateEnumKeyConverter(keyType);
+                // Ignore failure case here in multi-threaded cases since the cached item will be equivalent.
+                s_keyConverters.TryAdd(keyType, converter);
```

Comment:
Use the helper `TryAdd()` in JsonHelpers -- that will fix your build breaks against netstandard.

Reply:
I switched `s_keyConverters` to a `ConcurrentDictionary` and hence the issue was fixed.